tests/integration/ps.bats

```diff
@@ -3,18 +3,21 @@
 load helpers
 
 function setup() {
+	# ps requires cgroups
+	[ $EUID -ne 0 ] && requires rootless_cgroup
+
 	setup_busybox
+	if [ $EUID -ne 0 ]; then
```

Comment:
```suggestion
[ $EUID -ne 0 ] && set_cgroups_path
```

or use `if` in L7 for consistency

Reply:
The reason is, bash has a peculiar syntax. commands that involve `test`, `[ ... ]`, `!` are not triggering `set -e` ("Exit immediately if a command exits with a non-zero status"), so something like `[ $EUID -ne 0 ] && set_cgroups_path` does not fail with `set -e` and when the condition is false.

Except the case when this is the last statement in the function, in which case its exit code becomes the exit code of the function.

As a result, something like this works:

```bash
#!/bin/bash

set -e

function foobar() {
    [ "$A" = "a" ] && B="b"
    echo "$A" "$B"
}

foobar
```

and something like this does not:

``````bash
#!/bin/bash

set -e

function foobar() {
    [ "$A" = "a" ] && B="b"
}

foobar # returns 1, so the script ends here
echo "$A" "$B"
```

Reply:
I mean,
```bash
[ $EUID -ne 0 ] && set_cgroups_path
```

at the end of the function will make the function return 1 if the condition inside `[ ]` is false.

Once the function returns 1, the script fails, because of `set -x`.

The workaround is to add `true` before the end of the function.